src/crawlee/_request.py

```diff
@@ -136,16 +135,8 @@ class BaseRequestData(BaseModel):
     headers: Annotated[HttpHeaders, Field(default_factory=HttpHeaders)] = HttpHeaders()
     """HTTP request headers."""
 
-    payload: Annotated[
-        HttpPayload | None,
-        BeforeValidator(lambda v: v.encode() if isinstance(v, str) else v),
```

Comment:
Isn't the `BeforeValidator` still needed when loading an old request file?

Reply:
This will work the same for old requests and new requests.

https://github.com/apify/crawlee-python/blob/c7cae423b3b09a5f69da1f6881565b179e014054/src/crawlee/memory_storage_client/_creation_management.py#L370

If it is an old file, we apply alias `json_` and deserialize the string.

I find it convenient to have this logic inside the model rather than in `_creation_management.py`

Reply:
Good point. We need both so that payload is correctly serialized and deserialized to a file.
Updated the e2e test.